### ISSUE TITLE
Relationship is settable via update mutation, but settable directive is set to false.

### ISSUE THREAD

**Issue by AccsoSG** [neo4j/graphql#5012]
**Describe the bug**
Setting a field with relationship is possible with an update mutation, although settable onUpdate is set to false.

**Type definitions**
````graphql
type Identifier {
  value: String!

  identifierType: IdentifierType! 
    @relationship(
      type: "HAS_IDENTIFIER_TYPE", 
      direction: OUT, 
      nestedOperations: [CONNECT, DISCONNECT],
      aggregate: false)
    @selectable(onRead: true, onAggregate: false)
    @settable(onCreate: true, onUpdate: false)
    @filterable(byValue: true, byAggregate: false)
}

type IdentifierType {
  name: String!
}
````

**To Reproduce**

1. Run a server with the following code...
2. Execute the following Mutation...
- Create
````graphql
mutation M1 {
  createIdentifierTypes(input: [{ name: "Test1" }, { name: "Test2" }]) {
    identifierTypes {
      name
    }
  }
}
````
````graphql
mutation M2 {
  createIdentifiers(input: {
    value: "123",
    identifierType: {
      connect: { where: { node: { name: "Test1"}}}
    }
  }){
    identifiers {
      value
      identifierType {
        name
      }
    }
  }
}
````

![image](https://github.com/neo4j/graphql/assets/49721200/17c80255-f172-45da-9a6f-c622d29e5da4)

- Update
This mutation is valid, but does not work because of the cardinality (Identifier.identifierType required exactly once):
````graphql

mutation M3 {
  updateIdentifiers(
    where: { value: "123" }
    connect: { identifierType: { where: { node: { name: "Test2" } } } }
  ) {
    identifiers {
      value
      identifierType {
        name
      }
    }
  }
}
````

This mutation, however, works.
````graphql
mutation M4 {
  updateIdentifiers(
    where: { value: "123" }
    connect: { identifierType: { where: { node: { name: "Test2" } } } },
    disconnect: { identifierType: { where: { node: { NOT: { name: "Test2"}}}}}
  ) {
    identifiers {
      value
      identifierType {
        name
      }
    }
  }
}
````

![image](https://github.com/neo4j/graphql/assets/49721200/a8161d58-7521-4535-8674-81d29ea2abf7)

**Expected behavior**
If the settable directive for onUpdate is set to false, the relationship should not be able to be changed. The generated schema should not offer this option at all.

**System (please complete the following information):**

* Version: @neo4j/graphql@5.3.2

**Side question**
If you set onUpdate to true via the settable directive, there is also the option of setting the relationship via the path 
````
update: { identifierType : { connect: { where: { node: { ... }}}}}
````
 to do exactly the same thing. Are these simply two different ways of doing the same thing, or is there a difference here?

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @AccsoSG. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by Liam-Doodson** [comment]
Hi @AccsoSG! I can see how this could be misleading due to the multiple inputs for performing the same action.

As you say, adding `@settable(onUpdate: false)` removes the option to update the related type using the following input `update: { identifierType : { connect: { where: { node: { ... }}}}}`. In other words, the field has been removed from the `identifierInput` input type. This is the expected behaviour for the directive and is documented [here](https://neo4j.com/docs/graphql/current/schema-configuration/field-configuration/#_usage_3).

This prevents users of the api from updating the values of the `identifierType` when coming from an `identifier`. If you also want to remove the option to connect/disconnect the relation to an `identifierType` when coming from an `identifier`, you'd need to remove the `CONNECT`/`DISCONNECT` from the `nestedOperations` argument of `@relationship`

**Comment by AccsoSG** [comment]
Hi @Liam-Doodson 
Thank you for your answer.

But if I remove Connect/Disconnect, then I can no longer set this relationship in a Create-Operation, can I?

````graphql
type Identifier {
  value: String!

  identifierType: IdentifierType! 
    @relationship(
      type: "HAS_IDENTIFIER_TYPE", 
      direction: OUT, 
      nestedOperations: [], #<----------------------------
      aggregate: false)
    @selectable(onRead: true, onAggregate: false)
    @settable(onCreate: true, onUpdate: false)
    @filterable(byValue: true, byAggregate: false)
}

type IdentifierType {
  name: String!
}
````

**Comment by Liam-Doodson** [comment]
Are you able to just remove the `DISCONNECT` `nestedOperation`? That way it would be possible to set the relationship but  it would not be possible to change the identifierType when coming from an identifier

**Comment by AccsoSG** [comment]
Yes, that would work, but there is still the corresponding CONNECT in the generated API. This CONNECT will not work, but it will still be visible.

My expectation of the settable directive in combination with the relationship directive is different...

**Comment by AccsoSG** [comment]
So perhaps it is more of a feature request. The settable directive should really also consider this case.

**Comment by Liam-Doodson** [comment]
As in you want to have the connect option on a create but not on an update mutation?

If this is the case I think it would be more appropriate to add this as an option to the nestedOperation argument potentially with an interface like this: `nestedOperation: { create: [CONNECT] }`

**Comment by AccsoSG** [comment]
I generally mean that the settable directive implies that a field cannot be set in the update case if onUpdate=false. But the fact is that this field can be set if it is a relationship. That is not very intuitive

I would not modify the nestedOperations of the relationship directive as you have suggested. I think that is too complicated. I would simply adapt the implementation of the settable directive so that the connect/disconnect options also disappears from the API if the settable directive is set to false.

**Comment by AccsoSG** [comment]
@Liam-Doodson I would be grateful if we could turn this ticket into a feature request instead of closing it.

**Comment by angrykoala** [comment]
Hi @AccsoSG 

Following some [breaking changes](https://neo4j.com/docs/graphql/current/migration/#_removed_top_level_arguments_that_are_not_update_from_the_top_level_update) to the API in version 6 this issue seems to be no longer relevant

The following mutation is no longer valid in version 6:
```graphql
mutation M4 {
  updateIdentifiers(
    where: { value: "123" }
    connect: { identifierType: { where: { node: { name: "Test2" } } } },
    disconnect: { identifierType: { where: { node: { NOT: { name: "Test2"}}}}}
  ) {
    identifiers {
      value
      identifierType {
        name
      }
    }
  }
}
```

To achieve the same behavior, the following query must be used (note that `connect` and `disconnect` are now inside `update`):

```graphql
mutation M4 {
  updateIdentifiers(
    where: { value: "123" }
    update: {
      identifierType: [
        {
          connect: { where: { node: { name: "Test2" } } }
          disconnect: { where: { node: { NOT: { name: "Test2" } } } }
        }
      ]
    }
  ) {
    identifiers {
      value
      identifierType {
        name
      }
    }
  }
}
```

As discussed in this thread, the field `identifierType` is not available in `update`, meaning that connect and disconnect are no longer valid if `settable` is false, as expected in this issue

Let me know if you have any questions or if I missed anything

**Comment by angrykoala** [comment]
An edge case for nested connect still happens, this has been reported on the follow up issue #6118
